core/app/models/spree/promotion.rb

```diff
@@ -84,7 +93,7 @@ def not_expired?
     end
 
     def active?
-      started? && not_expired?
+      started? && not_expired? && (Spree::Config.actionless_promotion_inactive ? actions.present? : true)
```

Comment:
What about a different method (`has_actions?`) for this?

Reply:
I think that `has_actions?` would be confused because it will return `true` is the preference `actionless_promotion_inactive` is false.

I have considered this suggestion and following the @aldesantis comments I have named it `consider_active_by_actions_presence`